Documentation/container-lifecycle.md

```diff
@@ -0,0 +1,34 @@
+# Lifecycle of a container in rocket
+
+## Creation
+
+`rkt run` instantiates a container uuid and filesystem tree rooted at `/var/lib/rkt/containers/$uuid/.`, henceforth referred to as `$cdir`.
+
+An exclusive advisory lock is acquired on `$cdir` with a file descriptor associated with the process executing the stage1 /init.  This lock is automatically released when the /init process exits and the kernel closes all of the remaining open file descriptors.
+
+
+## Death / exit
+
+A container is considered exited if an advisory lock can be acquired on `$cdir`.  Upon exit of a container's /init process, the exclusive `$cdir` lock acquired at the beginning of `rkt run` becomes released by the kernel.
+
+## Garbage collection
+
+Exited containers are discarded in a common gc mark-and-sweep style by invoking the `rkt gc` command:
```

Comment:
For people that aren't familiar with mark-and-sweep, is there a particular reason we're doing it like that?